### ISSUE TITLE
A bad manual correction of the Style/SafeNavigation cop isn't raised as now redundant

### ISSUE THREAD

**Issue by mkissarli** [rubocop/rubocop#10926]
This request came about due to an incorrectly formatted piece of code which followed the pattern:

```
a && a.foo
```

This was correctly flagged up as an offence of `Style/SafeNavigation`, and so was manually adjusted to:

```
a && a&.foo
```

which was not flagged up any further, despite the now redundant `nil` check.

## Describe the solution you'd like

I think in this situation there should be a further offence flagged up for the now redundant `nil` check. For example:

```
a && a&.foo
```

should cause a `Style/RedundantNilCheck` flag (or something similar), which itself can be fixed with: 

```
a&.foo
```

## Describe alternatives you've considered

Writing a custom cop for this, although I think it would make a good default.
